apps/emqx_schema_registry/src/emqx_schema_registry_serde.erl

```diff
@@ -64,7 +64,21 @@ handle_rule_function(sparkplug_encode, [Term | MoreArgs]) ->
         [?EMQX_SCHEMA_REGISTRY_SPARKPLUGB_SCHEMA_NAME, Term | MoreArgs]
     );
 handle_rule_function(schema_decode, [SchemaId, Data | MoreArgs]) ->
-    decode(SchemaId, Data, MoreArgs);
+    try
+        decode(SchemaId, Data, MoreArgs)
+    catch
+        error:{gpb_error, {decoding_failure, {_Data, _Schema, {error, function_clause, _Stack}}}} ->
```

Comment:
this is very much prone to missing code coverage after the lib upgrade etc.
please add a test case which asserts this exact throw for regression purpose.

Reply:
Good point. Will do.

Reply:
Fixed in [9b089a4](https://github.com/emqx/emqx/pull/13147/commits/9b089a4e5aa44c3013743a325002c21d1a0e5bc0)